client/src/components/BeginPasswordReset/BeginPasswordReset.js

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import './BeginPasswordReset.scss';
+import { validEmailRegex } from '../../utils/string-util';
+import { Card, Form } from 'react-bootstrap';
+import Logo from '../Logo/Logo';
+import Button from 'react-bootstrap/Button';
+
+export default class BeginPasswordReset extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            validated: false,
+            email: '',
+            errors: {
+                email: ''
+            }
+        };
+    }
+
+    validateForm(errors) {
+        let valid = !(errors.email.length > 0);
+        this.handleChange({
+            target: {
+                name: 'email',
+                value: this.state.email
+            }
+        });
+        this.setState({validated: true});
+        return valid;
+    }
+
+    handleSubmit(event) {
+        const { onResetPassword } = this.props;
+        const { errors, email } = this.state;
+        event.preventDefault();
+        if (this.validateForm(errors)) {
+            onResetPassword(email);
+        }
+    }
+
+    handleChange(event) {
+        const { errors } = this.state;
+        const { name, value } = event.target;
+        if (event.preventDefault) event.preventDefault();
+
+        if (name === 'email') {
+            errors.email = validEmailRegex.test(value)
+                ? ''
+                : 'Email is not valid';
+        }
+
+        this.setState({errors, [name]: value})
+    }
```

Comment:
Any reason to have this abstracted out beyond validating emails?

Reply:
Nope, this was copy and pasted from somewhere else with multiple fields. Fixed now